http-client/Network/HTTP/Client/Types.hs

```diff
@@ -166,6 +166,7 @@ data HttpExceptionContent
                    -- @since 0.5.0
                    | InvalidHeader S.ByteString
                    -- ^ The given response header line could not be parsed
+                   --   or given request header is not compliant (e.g. has newlines)
```

Comment:
I don't like abusing an existing constructor for a new meaning.

Reply:
I understand what you mean, but this particular case seems like a perfect fit, InvalidHeader for invalid headers. Anyway, I don't feel strongly about this.